### PR TITLE
how-to/troubleshoot: add TiDB diagnose map

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Add TiDB diagnose map.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-map.md

### CODE REVIEW

**Comment by ran-huang** [comment]
Five deadlinks in the original file: [case-517](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case517.md), [case-1004](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case1004.md), [case-275](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case275.md), [case-852](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case852.md), [case-764](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case764.md).

**Comment by ran-huang** [comment]
@shenli FYI.

**Comment by ran-huang** [comment]
@lilin90 Comments addressed. Also updated links and format. PTAL~

File: how-to/troubleshoot/diagnose-map.md [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+
+    - 原因 3：早期版本（v2.1.15 & v3.0.0-rc1 之前）PD 实例删除 TiDB key 无效的问题，会导致每次 DDL 变更都需要等 2 个 lease（很慢）。
+
+    - 其他未知原因，请[上报 bug](https://github.com/pingcap/tidb/issues/new?labels=type%2Fbug&template=bug-report.md)。
+
+    - 解决方法：原因 1 需要检查与外部组件的网络问题；原因 2 和 3 已经修复，需要升级到高版本；其他原因，可选择以下兜底方案进行 DDL owner 迁移。
+
+    - DDL owner 迁移方案：
+
+        - 如果与该台 TiDB 可以网络互通，执行重新进行 owner 选举命令：`curl -X POST http://{TiDBIP}:10080/ddl/owner/resign`
```
[the remark this answers is not in the document]
Comment (ran-huang) [reply how-to/troubleshoot/diagnose-map.md:71]: @zhangjinpeng1987 PTAL

File: how-to/troubleshoot/diagnose-map.md [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+
+    - v3.0 及以上版本可以使用 `SQL Bind` 功能固定执行计划。
+
+    - 更新统计信息。在大致确定问题是由统计信息导致的情况下，先 [dump 统计信息](https://pingcap.com/docs-cn/stable/reference/performance/statistics/#%E5%AF%BC%E5%87%BA%E7%BB%9F%E8%AE%A1%E4%BF%A1%E6%81%AF)保留现场。如果是由于统计信息过期导致，例如 `show stats_meta` 中 modify count/row count 大于某个值（例如 0.3）或者表中存在时间列的索引情况下，可以先尝试 analyze table 恢复；如果配置了 auto analyze，可以查看系统变量 `tidb_auto_analyze_ratio` 是否过大（例如大于 0.3），以及当前时间是否在 `tidb_auto_analyze_start_time` 和 `tidb_auto_analyze_end_time` 范围内。
+
+    - 其他情况，请[上报 bug](https://github.com/pingcap/tidb/issues/new?labels=type%2Fbug&template=bug-report.md)。
+
+### 3.4 SQL 执行报错
+
+- 3.4.1 客户端报 `ERROR 1265(01000) Data Truncated` 错误。原因是 TiDB 内在计算 `Decimal` 类型处理精度的时候，和 MySQL 不兼容（内部正在修复中，预计 v3.0.10 版本修复），具体原因如下：
```
[the remark this answers is not in the document]
Comment (ran-huang) [reply how-to/troubleshoot/diagnose-map.md]: Updated.

File: how-to/troubleshoot/diagnose-map.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (AilinKid) [reply how-to/troubleshoot/diagnose-map.md:71]: TiDB 集群

[the remark this answers is not in the document]
Comment (AilinKid) [reply how-to/troubleshoot/diagnose-map.md:71]: TiDB 集群

File: how-to/troubleshoot/diagnose-map.md [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+
+    - 原因 3：早期版本（v2.1.15 & v3.0.0-rc1 之前）PD 实例删除 TiDB key 无效的问题，会导致每次 DDL 变更都需要等 2 个 lease（很慢）。
+
+    - 其他未知原因，请[上报 bug](https://github.com/pingcap/tidb/issues/new?labels=type%2Fbug&template=bug-report.md)。
+
+    - 解决方法：原因 1 需要检查与外部组件的网络问题；原因 2 和 3 已经修复，需要升级到高版本；其他原因，可选择以下兜底方案进行 DDL owner 迁移。
+
+    - DDL owner 迁移方案：
+
+        - 如果与该台 TiDB 可以网络互通，执行重新进行 owner 选举命令：`curl -X POST http://{TiDBIP}:10080/ddl/owner/resign`
```
Comment (ran-huang) [inline how-to/troubleshoot/diagnose-map.md:71]:
```suggestion
        - 如果与该 TiDB 集群可以网络互通，执行重新进行 owner 选举命令：`curl -X POST http://{TiDBIP}:10080/ddl/owner/resign`
```

File: how-to/troubleshoot/diagnose-map.md [hunk 4]
```diff
@@ -0,0 +1,549 @@
...
+
+        - 如果与该台 TiDB 不可以网络互通，需旁路下线，通过 `tidb-ctl` 工具，从 PD 集群的 etcd 中直接删除 DDL owner，之后也会重新选举：`tidb-ctl etcd delowner [LeaseID] [flags] + ownerKey`
```
Comment (ran-huang) [inline how-to/troubleshoot/diagnose-map.md:73]:
```suggestion
        - 如果与该 TiDB 集群不可以网络互通，需旁路下线，通过 `tidb-ctl` 工具，从 PD 集群的 etcd 中直接删除 DDL owner，之后也会重新选举：`tidb-ctl etcd delowner [LeaseID] [flags] + ownerKey`
```

**Comment by ran-huang** [comment]
@TomShawn PTAL again~ Thanks!

**Pull request merged**